promql/functions.go

```diff
@@ -1347,11 +1333,8 @@ func funcHistogramAvg(vals []parser.Value, _ parser.Expressions, enh *EvalNodeHe
 	return enh.Out, nil
 }
 
-// === histogram_stddev(Vector parser.ValueTypeVector) (Vector, Annotations)  ===
-func funcHistogramStdDev(vals []parser.Value, _ parser.Expressions, enh *EvalNodeHelper) (Vector, annotations.Annotations) {
-	inVec := vals[0].(Vector)
-
-	for _, sample := range inVec {
+func histogramVariance(vec Vector, enh *EvalNodeHelper, varianceToResult func(float64) float64) (Vector, annotations.Annotations) {
+	for _, sample := range vec {
```

Comment:
Can't we save the callers the hassle of converting `vals[0]` to a `Vector`? Seems like unnecessary duplication to me.
```suggestion
func histogramVariance(vals []parser.Value, enh *EvalNodeHelper, varianceToResult func(float64) float64) (Vector, annotations.Annotations) {
	vec := vals[0].(Vector)
	for _, sample := range vec {
```

Reply:
Yeah, we can do that. I'll add this change as well.